tamr_client/dataset/__init__.py

```diff
@@ -0,0 +1,7 @@
+# flake8: noqa
+from tamr_client.dataset.dataset import Dataset
+
+from tamr_client.dataset.dataset import from_resource_id
+from tamr_client.dataset.dataset import DatasetNotFound
```

Comment:
`DatasetNotFound` should also be renamed (here and in `dataset.py`) to make sure of namespacing:

```suggestion
from tamr_client.dataset.dataset import NotFound
```

You'll likely need to change this in a couple tests and in the docs too.